src/main/java/Node.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * A Node class for a singly linked structure. Each node
+ * contains a reference to data of type T, which may be
+ * null, and a reference to the next/subsequent/successor
+ * singly linked node, which may also be null.
+ *
+ * @param <T>   Type of the data being stored in the node
+ */
+public class Node<T> {
```

Comment:
> Originally this was not going to contain the documentation, but I threw it in anyways.

I'd argue this class should not be `public`. It's an implementation detail of the linked list, classes outside will never need access to it and don't need to know it exists.

Semi related to #46 but I'd probably make some kind of linked list package and make this a package private class. Or, better yet, this should be a static inner class of the linked list class (it doesn't deserve to be top level (the [jdk agrees](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/LinkedList.java#L974))). This will also help reduce your namespace pollution. What are you going to call a doubly linked node if it has to live in this package? A tree node?

Reply:
> Or, better yet, this should be a static inner class of the linked list class...

I love this idea. 

I think for now I will open an issue and then do the switch when creating `LinkedStack`. That way it will give a chance to reflect on design.